internal/controllers/machinedeployment/suite_test.go

```diff
@@ -88,18 +89,25 @@ func TestMain(m *testing.M) {
 		}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
 			panic(fmt.Sprintf("Failed to start ClusterCacheReconciler: %v", err))
 		}
+		if err := (&machinecontroller.Reconciler{
+			Client:    mgr.GetClient(),
+			APIReader: mgr.GetAPIReader(),
+			Tracker:   tracker,
+		}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+			panic(fmt.Sprintf("Failed to start MachineReconciler: %v", err))
+		}
```

Comment:
Any particular reason why this was added, assuming you meant to add a controller test, is it?

Reply:
👋 to address [this failing test case](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/8463/pull-cluster-api-test-main/1643160008172507136) where the `MachineSet` that was being replaced was[ failing to scale down](https://github.com/kubernetes-sigs/cluster-api/blob/333fd2c3cdb6de047d345389c87eeda0986091c3/internal/controllers/machinedeployment/machinedeployment_controller_test.go#L408-L432).  

Seems that this test was dependent on the less than desired behavior described in the linked issue in order to pass, as the `MachineSet` controller could simply[ issue the delete request](https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/machineset/machineset_controller.go#L571) and the command would succeed.  

Now that the finalizer is added on `Machine` creation, the `TestMachineDeploymentReconciler` test requires the `Machine` controller to be running in the test suite, so it can step through [ReconcileDelete](https://github.com/kubernetes-sigs/cluster-api/blob/333fd2c3cdb6de047d345389c87eeda0986091c3/internal/controllers/machine/machine_controller.go#L306), remove the finalizer and the `MachineSet` can successfully scale down.

Reply:
@srm09 does that make sense to you?  I have also added a check in the existing `TestComputeDesiredMachine` test to ensure the expected `Finalizer` is present on the output `Machine`

Reply:
hey @srm09 wanted to bump this and see if you have any additional feedback on this PR

Reply:
Sorry for the late reply on this. The change makes sense. However, I am wondering if it would be better to just drop the finalizer form the machines in this test over adding the entire machine reconciler here. In this test we are already patching to machines with fake node refs to mimic what a machine controller would do, it might be simpler to also include to a patch to drop the finalizer on the machine along with it. 

I don't have strong opinions and I am fine either way.

@sbueringer WDYT?

Reply:
no worries, I see what you're saying -- will wait for further updates on this thread before making any additional changes

Reply:
Will try to take a look soon